Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/VaccinatedPerson.kt

```diff
@@ -71,6 +71,16 @@ data class VaccinatedPerson(
         return IMMUNITY_WAITING_DAYS - Days.daysBetween(newestFullDose.vaccinatedOn, today).days
     }
 
+    private fun getNewestFullDose() = vaccinationCertificates.filter { it.doseNumber == it.totalSeriesOfDoses }
+
+    fun isFirstVaccinationDoseAfterRecovery(): Boolean {
+        val vaccinationDetails = getNewestFullDose()[0].rawCertificate.vaccination
```

Comment:
- Please add tests? 
- are you sure that is not causing an exception for other vaccine doses 1,2 or maybe even 3 in future 
`getNewestFullDose` is looking only for certificate where  `doseNumber == totalSeriesOfDoses` if Certificate is 1 of 2, 1 of 3, or 2 of 3 `getNewestFullDose` will return an emptyList and `getNewestFullDose()[0]` might cause a crash ??

Reply:
Yes, you're right, I'm working on it right now.